### PR TITLE
Rss ecomm 3 14

### PR DESCRIPTION
**Pull request by vetalapo**
1. Task: [link](https://github.com/rolling-scopes-school/tasks/blob/master/tasks/eCommerce-Application/Sprints/Sprint3/RSS-ECOMM-3_14.md)
2. Screenshot:
image

3. Change description:

Made user profile page which displays user's full name, birthday, addresses (regular, types, and default)
Updated routing so only logged in users can see their profiles.

### CODE REVIEW

**Review by valeryiay — CHANGES_REQUESTED** [review]
I've added a few comments about formatting, but the PR looks good overall.

File: src/views/ProfileView.vue [hunk 1]
```diff
@@ -0,0 +1,198 @@
...
+                        title: "Country",
+                        key: "country",
+                        value: (item: FullCustomerAddress) => {
+                            const country = COUNTRIES.find((c) => c.code == item.country);
+
+                            if (!country) {
+                                return item.country;
+                            }
+
+                            return country.title
```
[the remark this answers is not in the document]
Comment (vetalapo) [reply src/views/ProfileView.vue:48]: Good catch. 👌

File: src/views/ProfileView.vue [hunk 2]
```diff
@@ -0,0 +1,198 @@
...
+                        key: "address",
+                        value: (item: FullCustomerAddress) =>
+                            `${item.streetName} ${item.streetNumber}`
+                    },
+                    { title: "Postal Code", value: "postalCode" },
+                    {
+                        title: "Address Type",
+                        key: "addressType",
+                        value: (item: FullCustomerAddress) => {
+                            const isShippingAddress = this.authStore?.user?.user?.shippingAddressIds.includes(item.id);
```
[the remark this answers is not in the document]
Comment (vetalapo) [reply src/views/ProfileView.vue:68]: 72, 79 - it doesn't matter, but ok.

What's wrong with 74, 81, 83?

File: src/views/ProfileView.vue [hunk 3]
```diff
@@ -0,0 +1,198 @@
...
+                                return `Billing${isBillingAddressDefault ? " (Default)" : ""}`;
+                            }
+
+                            return "";
+                        }
+                    }
+                ];
+            }
+        }
+    }
```
[the remark this answers is not in the document]
Comment (vetalapo) [reply src/views/ProfileView.vue:88]: 👍

File: src/views/ProfileView.vue [hunk 4]
```diff
@@ -0,0 +1,198 @@
...
+                                    color="grey-darken-3"
+                                    size="200"
+                                    rounded="0"
+                                >
+                                    <img
+                                        height="100%"
+                                        min-width="200"
+                                        alt="Default profile image"
+                                        src="@/assets/img/default-avatar.svg"
+                                    >
```
[the remark this answers is not in the document]
Comment (vetalapo) [reply src/views/ProfileView.vue:114]: It's self closing. And it's actually a bad practice to close this tags as /> in HTML
https://jakearchibald.com/2023/against-self-closing-tags-in-html/

File: src/views/ProfileView.vue [hunk 5]
```diff
@@ -0,0 +1,198 @@
...
+                                </v-avatar>
+                            </v-card>
+                        </v-col>
+                        <v-col>
+                            <h2 class="text-h4 font-weight-black">{{ authStore?.user?.user?.firstName }} {{authStore?.user?.user?.lastName}}</h2>
```
[the remark this answers is not in the document]
Comment (vetalapo) [reply src/views/ProfileView.vue:119]: 125, 164? What's wrong? They look okay.

File: src/views/ProfileView.vue [hunk 6]
```diff
@@ -0,0 +1,198 @@
...
+    .v-row {
+        margin: 0;
+    }
+
+    .customer-avatar-column {
+        display: inline-table;
+    }
+
+    .customer-avatar {
+        border-radius: 5%!important;
```
[the remark this answers is not in the document]
Comment (vetalapo) [reply src/views/ProfileView.vue:191]: It's doesn't help with anything but ok 👍

File: src/views/RootView.vue [hunk 7]
```diff
@@ -5,10 +5,19 @@
...
+        },
+        computed: {
+            getFullCustomerName(): string {
+                const firstName = this.authStore?.user?.user?.firstName;
+                const lastName = this.authStore?.user?.user?.lastName;
+
+                return `${firstName} ${lastName}`;
+            }
         }
     }
```
[the remark this answers is not in the document]
Comment (vetalapo) [reply src/views/RootView.vue:22]: 👍

File: src/views/RootView.vue [hunk 8]
```diff
@@ -35,48 +44,58 @@
...
+                                class="mx-auto d-flex align-center"
+                                v-bind="props"
+                            >
+                                <v-avatar color="grey-darken-3 ml-1">
+                                    <img
+                                        width="44"
+                                        height="44"
+                                        alt="Default profile image"
+                                        src="@/assets/img/default-avatar.svg"
+                                    >
```
[the remark this answers is not in the document]
Comment (vetalapo) [reply src/views/RootView.vue:61]: It's self closing. And it's actually a bad practice to close this tags as /> in HTML
https://jakearchibald.com/2023/against-self-closing-tags-in-html/

File: src/views/RootView.vue [hunk 9]
```diff
@@ -95,6 +114,11 @@
         margin-right: 250px;
```
[the remark this answers is not in the document]
Comment (vetalapo) [reply src/views/RootView.vue:114]: 👍

File: src/views/__tests__/ProfileView.spec.ts [hunk 10]
```diff
@@ -0,0 +1,30 @@
+import { mount, VueWrapper } from "@vue/test-utils";
+import { beforeEach, describe, it, expect, vi } from "vitest";
+import vuetify from "@/plugins/vuetify";
+import router from "@/router";
+import ProfileView from "../ProfileView.vue";
+import { useAuthStore } from "@/store";
```
[the remark this answers is not in the document]
Comment (vetalapo) [reply src/views/__tests__/ProfileView.spec.ts:6]: It'll be needed for the next PR. I think I'm going to leave it there for now.

File: src/views/RegistrationView.vue
Comment (valeryiay) [inline src/views/RegistrationView.vue:1]: double quotes on 80 line

**Review by valeryiay — APPROVED** [review]
looks good. thank you!

**Review by JoMarianek — COMMENTED** [review]
Looks good good, thank you!

File: src/views/RegistrationView.vue
Comment (vetalapo) [reply src/views/RegistrationView.vue:1]: 👍

**Pull request merged**